src/main/java/com/github/bhlangonijr/chesslib/pgn/PgnHolder.java

```diff
@@ -157,161 +157,181 @@ public void loadPgn() throws Exception {
                         if (p != null) {
                             String tag = p.name.toLowerCase().trim();
                             //begin
-                            if (tag.equals("event")) {
-                                if (moveTextParsing && moveText != null && game != null &&
-                                        game.getHalfMoves().size() == 0) {
-                                    setMoveText(game, moveText);
-                                }
-                                size++;
-                                for (PgnLoadListener l : getListener()) {
-                                    l.notifyProgress(size);
-                                }
-                                game = null;
-                                round = null;
-                                whitePlayer = null;
-                                blackPlayer = null;
-                                event = getEvent().get(p.value);
-                                if (event == null) {
-                                    event = GameFactory.newEvent(p.value);
-                                    event.setPgnHolder(this);
-                                    getEvent().put(p.value, event);
-                                }
-                                moveText = new StringBuilder();
+                            switch (tag) {
```

Comment:
A big part of these changes have been added in another PR. There's a lot of conflicts with master and this PR needs an update before merging.